src/simple_lottery/init.rs

```diff
@@ -0,0 +1,94 @@
+//! Parameter derivation for simple lottery.
+
+use super::params::Params;
+use super::setup::Setup;
+
+use std::f64::consts::LN_2;
+
+/// Calculates Setup parameters.
+pub fn make_setup(params: &Params) -> Setup {
+    let ratio = params.set_size as f64 / params.lower_bound as f64;
+
+    // Execute binary search for the optimal value of `rs`.
+
+    let mut l: f64 = 0.0;
+    let mut r: f64 = ratio;
+
+    loop {
+        let m = (l + r) / 2.0;
+        let rs = m;
+        let rc = ratio / rs;
+
+        let lhs = params.soundness_param / (rs.ln() - 1.0 + 1.0 / rs);
```

Comment:
`soundness_bound` then? Any variant of it to make it more descriptive and to make it more distinctive from the `l` and `r` afterwards would be great.

Reply:
The name `lhs` only stands for "left hand side". Why left? Is it simply because it is written on the left of the bound check in the paper or for some other reason? I do not know.

At the moment, `lhs` is less descriptive and meaningful than `soundness_bound`. This value moreover is still a bound for `u,` perhaps loose, right, perhaps not sufficient but still necessary. If this is a problem, we can always multiply it by `ln2`.